lib/media/streaming_engine.js

```diff
@@ -329,14 +333,51 @@ shaka.media.StreamingEngine = class {
    *   reset the timestamp offset to ensure the new track segments are correctly
    *   placed on the timeline.
    */
-  switchVariant(
-      variant, clearBuffer = false, safeMargin = 0, force = false,
-      adaptation = false) {
+  async switchVariant(variant, clearBuffer = false,
+      safeMargin = 0, force = false, adaptation = false) {
+    this.previousVariant_ = this.currentVariant_;
     this.currentVariant_ = variant;
 
     if (!this.startupComplete_) {
       // The selected variant will be used in start().
       return;
+    } else {
+      // Audio codec change type
+      const ContentType = shaka.util.ManifestParserUtils.ContentType;
+      const Strategy = shaka.config.CodecSwitchingStrategy;
+      const changeRequired = (this.previousVariant_.audio.codecs !==
+        this.currentVariant_.audio.codecs);
+      const changeStrategy = this.config_.codecSwitchingStrategy;
+      const fullMimeType = shaka.util.MimeUtils.getFullType(
+          variant.audio.mimeType, variant.audio.codecs);
+
+      if (changeRequired) {
+        shaka.log.debug('Audio Codec Switch Required, using', changeStrategy);
+        switch (changeStrategy) {
+          case Strategy.DISABLED:
+            shaka.log.debug('Strategy DISABLED');
+            break;
```

Comment:
It also might make more sense for this to be an error.
By this point in the method, we have already set `this.currentVariant_` to the new variant, right? Things are going to assume that the variant has been switched. If it turns out that we would need to switch audio codecs to get here but audio codec switching is disabled, then something has gone wrong, right?